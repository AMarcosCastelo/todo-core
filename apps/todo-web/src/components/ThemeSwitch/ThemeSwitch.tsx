import { Switch, type SwitchProps } from '@nextui-org/react';

import MoonIcon from '../MoonIcon/MoonIcon';
import SunIcon from '../SunIcon/SunIcon';

interface Props extends Omit<Partial<SwitchProps>, 'iconOn' | 'iconOff'> {}

const ThemeSwitch: React.FC<Props> = (props) => {
  return (
    <Switch
      {...props}
      iconOn={<MoonIcon filled />}
      iconOff={<SunIcon filled />}
    />
  );
};

export default ThemeSwitch;
