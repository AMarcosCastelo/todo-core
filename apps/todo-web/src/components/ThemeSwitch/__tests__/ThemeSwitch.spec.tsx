import { describe, expect, it, vi } from 'vitest';

import { baseRender } from '../../../utils';
import ThemeSwitch from '../ThemeSwitch';

describe('ThemeSwitch', () => {
  it('should render correctly', () => {
    const screen = baseRender(<ThemeSwitch />);

    expect(screen).toMatchSnapshot();
  });

  it('should render SunIcon if switch was unchecked', () => {
    const screen = baseRender(<ThemeSwitch />);

    expect(screen.getByTestId('sun-filled-icon')).toBeInTheDocument();
    expect(screen).toMatchSnapshot();
  });

  it('should render MoonIcon if switch was checked', () => {
    const screen = baseRender(<ThemeSwitch checked={true} />);

    expect(screen.getByTestId('moon-filled-icon')).toBeInTheDocument();
    expect(screen).toMatchSnapshot();
  });
});
