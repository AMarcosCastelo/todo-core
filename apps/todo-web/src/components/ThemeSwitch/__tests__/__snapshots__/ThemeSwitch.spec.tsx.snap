// Vitest Snapshot v1

exports[`ThemeSwitch > should render MoonIcon if switch was checked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <label
          class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
          data-state="checked"
        >
          <input
            class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
            data-state="checked"
            tabindex="-1"
            type="checkbox"
          />
          <div
            aria-checked="true"
            aria-disabled="false"
            class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-bDUxer-checked-true nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--checked nextui-switch-checked"
            data-state="checked"
            role="switch"
            tabindex="0"
          >
            <span
              class="nextui-c-eJPKci nextui-switch-circle"
            >
              <svg
                data-testid="moon-filled-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M21.53 15.93c-.16-.27-.61-.69-1.73-.49a8.46 8.46 0 01-1.88.13 8.409 8.409 0 01-5.91-2.82 8.068 8.068 0 01-1.44-8.66c.44-1.01.13-1.54-.09-1.76s-.77-.55-1.83-.11a10.318 10.318 0 00-6.32 10.21 10.475 10.475 0 007.04 8.99 10 10 0 002.89.55c.16.01.32.02.48.02a10.5 10.5 0 008.47-4.27c.67-.93.49-1.519.32-1.79z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <label
        class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
        data-state="checked"
      >
        <input
          class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
          data-state="checked"
          tabindex="-1"
          type="checkbox"
        />
        <div
          aria-checked="true"
          aria-disabled="false"
          class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-bDUxer-checked-true nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--checked nextui-switch-checked"
          data-state="checked"
          role="switch"
          tabindex="0"
        >
          <span
            class="nextui-c-eJPKci nextui-switch-circle"
          >
            <svg
              data-testid="moon-filled-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M21.53 15.93c-.16-.27-.61-.69-1.73-.49a8.46 8.46 0 01-1.88.13 8.409 8.409 0 01-5.91-2.82 8.068 8.068 0 01-1.44-8.66c.44-1.01.13-1.54-.09-1.76s-.77-.55-1.83-.11a10.318 10.318 0 00-6.32 10.21 10.475 10.475 0 007.04 8.99 10 10 0 002.89.55c.16.01.32.02.48.02a10.5 10.5 0 008.47-4.27c.67-.93.49-1.519.32-1.79z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ThemeSwitch > should render SunIcon if switch was unchecked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <label
          class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
          data-state="unchecked"
        >
          <input
            class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
            data-state="unchecked"
            tabindex="-1"
            type="checkbox"
          />
          <div
            aria-checked="false"
            aria-disabled="false"
            class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--unchecked"
            data-state="unchecked"
            role="switch"
            tabindex="0"
          >
            <span
              class="nextui-c-eJPKci nextui-switch-circle"
            >
              <svg
                data-testid="sun-filled-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M19 12a7 7 0 11-7-7 7 7 0 017 7z"
                  />
                  <path
                    d="M12 22.96a.969.969 0 01-1-.96v-.08a1 1 0 012 0 1.038 1.038 0 01-1 1.04zm7.14-2.82a1.024 1.024 0 01-.71-.29l-.13-.13a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.984.984 0 01-.7.29zm-14.28 0a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a1 1 0 01-.7.29zM22 13h-.08a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zM2.08 13H2a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zm16.93-7.01a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a.984.984 0 01-.7.29zm-14.02 0a1.024 1.024 0 01-.71-.29l-.13-.14a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.97.97 0 01-.7.3zM12 3.04a.969.969 0 01-1-.96V2a1 1 0 012 0 1.038 1.038 0 01-1 1.04z"
                  />
                </g>
              </svg>
            </span>
          </div>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <label
        class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
        data-state="unchecked"
      >
        <input
          class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
          data-state="unchecked"
          tabindex="-1"
          type="checkbox"
        />
        <div
          aria-checked="false"
          aria-disabled="false"
          class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--unchecked"
          data-state="unchecked"
          role="switch"
          tabindex="0"
        >
          <span
            class="nextui-c-eJPKci nextui-switch-circle"
          >
            <svg
              data-testid="sun-filled-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M19 12a7 7 0 11-7-7 7 7 0 017 7z"
                />
                <path
                  d="M12 22.96a.969.969 0 01-1-.96v-.08a1 1 0 012 0 1.038 1.038 0 01-1 1.04zm7.14-2.82a1.024 1.024 0 01-.71-.29l-.13-.13a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.984.984 0 01-.7.29zm-14.28 0a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a1 1 0 01-.7.29zM22 13h-.08a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zM2.08 13H2a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zm16.93-7.01a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a.984.984 0 01-.7.29zm-14.02 0a1.024 1.024 0 01-.71-.29l-.13-.14a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.97.97 0 01-.7.3zM12 3.04a.969.969 0 01-1-.96V2a1 1 0 012 0 1.038 1.038 0 01-1 1.04z"
                />
              </g>
            </svg>
          </span>
        </div>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ThemeSwitch > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <label
          class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
          data-state="unchecked"
        >
          <input
            class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
            data-state="unchecked"
            tabindex="-1"
            type="checkbox"
          />
          <div
            aria-checked="false"
            aria-disabled="false"
            class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--unchecked"
            data-state="unchecked"
            role="switch"
            tabindex="0"
          >
            <span
              class="nextui-c-eJPKci nextui-switch-circle"
            >
              <svg
                data-testid="sun-filled-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M19 12a7 7 0 11-7-7 7 7 0 017 7z"
                  />
                  <path
                    d="M12 22.96a.969.969 0 01-1-.96v-.08a1 1 0 012 0 1.038 1.038 0 01-1 1.04zm7.14-2.82a1.024 1.024 0 01-.71-.29l-.13-.13a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.984.984 0 01-.7.29zm-14.28 0a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a1 1 0 01-.7.29zM22 13h-.08a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zM2.08 13H2a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zm16.93-7.01a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a.984.984 0 01-.7.29zm-14.02 0a1.024 1.024 0 01-.71-.29l-.13-.14a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.97.97 0 01-.7.3zM12 3.04a.969.969 0 01-1-.96V2a1 1 0 012 0 1.038 1.038 0 01-1 1.04z"
                  />
                </g>
              </svg>
            </span>
          </div>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <label
        class="nextui-c-STltJ nextui-c-STltJ-bwQcWP-color-default nextui-c-STltJ-JJjKI-size-md nextui-c-STltJ-bgIFDi-borderWeight-normal"
        data-state="unchecked"
      >
        <input
          class="nextui-c-jjKPqy nextui-c-iFHsrU nextui-c-jjKPqy-gwHlSF-animated-true nextui-switch-input"
          data-state="unchecked"
          tabindex="-1"
          type="checkbox"
        />
        <div
          aria-checked="false"
          aria-disabled="false"
          class="nextui-c-itiWTf nextui-c-huiNHE nextui-c-itiWTf-iXJPXp-animated-true nextui-switch nextui-switch--unchecked"
          data-state="unchecked"
          role="switch"
          tabindex="0"
        >
          <span
            class="nextui-c-eJPKci nextui-switch-circle"
          >
            <svg
              data-testid="sun-filled-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M19 12a7 7 0 11-7-7 7 7 0 017 7z"
                />
                <path
                  d="M12 22.96a.969.969 0 01-1-.96v-.08a1 1 0 012 0 1.038 1.038 0 01-1 1.04zm7.14-2.82a1.024 1.024 0 01-.71-.29l-.13-.13a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.984.984 0 01-.7.29zm-14.28 0a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a1 1 0 01-.7.29zM22 13h-.08a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zM2.08 13H2a1 1 0 010-2 1.038 1.038 0 011.04 1 .969.969 0 01-.96 1zm16.93-7.01a1.024 1.024 0 01-.71-.29 1 1 0 010-1.41l.13-.13a1 1 0 011.41 1.41l-.13.13a.984.984 0 01-.7.29zm-14.02 0a1.024 1.024 0 01-.71-.29l-.13-.14a1 1 0 011.41-1.41l.13.13a1 1 0 010 1.41.97.97 0 01-.7.3zM12 3.04a.969.969 0 01-1-.96V2a1 1 0 012 0 1.038 1.038 0 01-1 1.04z"
                />
              </g>
            </svg>
          </span>
        </div>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
